nrf_wifi/fw_if/umac_if/inc/fmac_tx.h

```diff
@@ -15,75 +11,163 @@
 #include "host_rpu_data_if.h"
 #include "fmac_structs.h"
 
+/**
+ * @defgroup fmac_tx FMAC TX
+ * @{
+ */
+
+/**
+ * @brief The maximum number of descriptors in a TX descriptor bucket.
+ */
 #define TX_DESC_BUCKET_BOUND 32
+
+/**
+ * @brief The length of the WMM parameters.
+ */
 #define DOT11_WMM_PARAMS_LEN 2
 
-/* 4 bits represent 4 access categories.
- * (VOVIBEBO) respectively
+/**
+ * @brief The size of the spare descriptor queue map.
+ *
+ * Four bits represent four access categories: VO, VI, BE, and BK.
  */
 #define SPARE_DESC_Q_MAP_SIZE 4
 
 /**
- * enum nrf_wifi_fmac_tx_status - The status of a TX operation performed by the
- *						RPU driver.
- * @NRF_WIFI_FMAC_TX_STATUS_SUCCESS: The TX operation was successful (sent packet to RPU).
- * @NRF_WIFI_FMAC_TX_STATUS_QUEUED: The TX operation was successful (packet queued in driver).
- * @NRF_WIFI_FMAC_TX_STATUS_FAIL: The TX operation failed.
- *
- * This enum lists the possible outcomes of a TX operation performed by the
- * RPU driver.
+ * @brief The status of a TX operation performed by the RPU driver.
+ */
+/**
+ * @brief The status of a TX operation performed by the RPU driver.
  */
 enum nrf_wifi_fmac_tx_status {
+	/** The TX operation was successful (sent packet to RPU). */
 	NRF_WIFI_FMAC_TX_STATUS_SUCCESS,
+	/** The TX operation was successful (packet queued in driver). */
 	NRF_WIFI_FMAC_TX_STATUS_QUEUED = 1,
-	NRF_WIFI_FMAC_TX_STATUS_FAIL = -1,
+	/** The TX operation failed. */
+	NRF_WIFI_FMAC_TX_STATUS_FAIL = -1
 };
 
+/**
+ * @brief Structure containing information about a TX packet.
+ */
 struct tx_pkt_info {
+	/** Pointer to the TX packet. */
 	void *pkt;
+	/** Peer ID. */
 	unsigned int peer_id;
 };
 
 #ifdef CONFIG_NRF700X_RAW_DATA_TX
+/**
+ * @brief Structure containing information for preparing a raw TX command.
+ */
 struct tx_cmd_prep_raw_info {
+	/** Pointer to the FMAC device context. */
 	struct nrf_wifi_fmac_dev_ctx *fmac_dev_ctx;
+	/** Pointer to the raw TX configuration. */
 	struct nrf_wifi_cmd_raw_tx *raw_config;
+	/** Number of TX packets. */
 	unsigned char num_tx_pkts;
 };
 #endif /* CONFIG_NRF700X_RAW_DATA_TX */
 
+/**
+ * @brief Structure containing information for preparing a TX command.
+ */
 struct tx_cmd_prep_info {
+	/** Pointer to the FMAC device context. */
 	struct nrf_wifi_fmac_dev_ctx *fmac_dev_ctx;
+	/** Pointer to the TX configuration. */
 	struct nrf_wifi_tx_buff *config;
 };
 
+/**
+ * @brief Initializes the TX module.
```

Comment:
```suggestion
 * @brief Initialize the TX module.
```
As mentioned before, use the imperative form of the verb in the function descriptions. In all places.

Reply:
Sorry, I did it for two files, but I guess it's everywhere, will fix. Thanks.